packages/js/components/src/experimental-select-control/select-control.tsx

```diff
@@ -190,6 +190,11 @@ function SelectControl< ItemType = DefaultItemType >( {
 						};
 					}
 					break;
+				case selectControlStateChangeTypes.ControlledPropUpdatedSelectedItem:
```

Comment:
This case was needed to resolve the `inputValue` not updating when the selected item changed. This was causing a bug with the `attribute-input-field` on the edit modal.